graylog2-server/src/main/java/org/graylog2/configuration/HttpConfiguration.java

```diff
@@ -0,0 +1,253 @@
+    public static final String PATH_WEB = "";
+    public static final String PATH_API = "api/";
+
+    @Parameter(value = "http_bind_address", required = true)
+    private HostAndPort httpBindAddress = HostAndPort.fromParts("127.0.0.1", GRAYLOG_DEFAULT_PORT);
+
+    @Parameter(value = "http_publish_uri", validator = URIAbsoluteValidator.class)
+    private URI httpPublishUri;
+
+    @Parameter(value = "http_enable_cors")
+    private boolean httpEnableCors = true;
+
+    @Parameter(value = "http_enable_gzip")
+    private boolean httpEnableGzip = true;
+
+    @Parameter(value = "http_max_header_size", required = true, validator = PositiveIntegerValidator.class)
+    private int httpMaxHeaderSize = 8192;
+
+    @Parameter(value = "http_thread_pool_size", required = true, validator = PositiveIntegerValidator.class)
+    private int httpThreadPoolSize = 16;
+
+    @Parameter(value = "http_selector_runners_count", required = true, validator = PositiveIntegerValidator.class)
+    private int httpSelectorRunnersCount = 1;
+
+    @Parameter(value = "http_enable_tls")
+    private boolean httpEnableTls = false;
+
+    @Parameter(value = "http_tls_cert_file")
+    private Path httpTlsCertFile;
+
+    @Parameter(value = "http_tls_key_file")
+    private Path httpTlsKeyFile;
+
+    @Parameter(value = "http_tls_key_password")
+    private String httpTlsKeyPassword;
+
+    @Parameter(value = "http_external_uri")
+    private URI httpExternalUri;
+
+    public HostAndPort getHttpBindAddress() {
+        return httpBindAddress
+                .requireBracketsForIPv6()
+                .withDefaultPort(9000);
+    }
+
+    public String getUriScheme() {
+        return getUriScheme(isHttpEnableTls());
+    }
+
+    public String getUriScheme(boolean enableTls) {
+        return enableTls ? "https" : "http";
+    }
+
+    public URI getHttpPublishUri() {
+        if (httpPublishUri == null) {
+            final URI defaultHttpUri = getDefaultHttpUri();
+            LOG.debug("No \"http_publish_uri\" set. Using default <{}>.", defaultHttpUri);
+            return defaultHttpUri;
+        } else if (WILDCARD_IP_ADDRESS.equals(httpPublishUri.getHost())) {
+            final URI defaultHttpUri = getDefaultHttpUri(httpPublishUri.getPath());
+            LOG.warn("\"{}\" is not a valid setting for \"http_publish_uri\". Using default <{}>.", httpPublishUri, defaultHttpUri);
+            return defaultHttpUri;
+        } else {
+            return Tools.normalizeURI(httpPublishUri, httpPublishUri.getScheme(), GRAYLOG_DEFAULT_PORT, httpPublishUri.getPath());
+        }
+    }
+
+    @VisibleForTesting
+    URI getDefaultHttpUri() {
+        return getDefaultHttpUri("/");
+    }
+
+    @VisibleForTesting
+    URI getDefaultHttpUri(String path) {
+        final HostAndPort bindAddress = getHttpBindAddress();
+
+        final URI publishUri;
+        if (WILDCARD_IP_ADDRESS.equals(bindAddress.getHost())) {
```

Comment:
This currently does not take the IPv6 wildcard address into account and will publish `[::]` as the URI, which is invalid.
I think that adding a check against `[::]` and extending `Tools.guessPrimaryNetworkAddress()` to also look for IPv6 addresses, would be the best option here.